OpenSim/Common/Test/testTRCFileAdapter.cpp

```diff
@@ -155,12 +155,12 @@ int main() {
         }
     }
 
-    std::cout << "Testing FileAdapter::readFile() and FileAdapter::writeFile()"
+    std::cout << "Testing FileAdapter::read() and FileAdapter::writeFile()"
               << std::endl;
     for(const auto& filename : filenames) {
         std::cout << "  " << filename << std::endl;
         try {
-            auto table = FileAdapter::readFile(filename).at("markers");
+            auto table = FileAdapter::createAdapterBasedOnExtension(filename)->read(filename).at("markers");
```

Comment:
Are there any other use cases for `createAdapterBasedOnExtension()`? This test is aimed at testing `read()`, but if there are other use cases it would be good to add them somewhere.

Reply:
For TRCFileAdapter there are no other use cases, you always create a table of concrete type from the file and TRCFileAdapter is used internally, I just didn't want to weaken the test.